Essentials/src/com/earth2me/essentials/commands/Commandtogglejail.java

```diff
@@ -38,29 +38,39 @@ public void run(final Server server, final CommandSource sender, final String co
                     return;
                 }
             }
+
             final User controller = sender.isPlayer() ? ess.getUser(sender.getPlayer()) : null;
             final JailStatusChangeEvent event = new JailStatusChangeEvent(player, controller, true);
             ess.getServer().getPluginManager().callEvent(event);
 
             if (!event.isCancelled()) {
+                final String jailName = args[1];
+                if (sender.isPlayer() && !ess.getUser(sender.getPlayer()).isAuthorized("essentials.togglejail." + jailName)) {
```

Comment:
This won't break everyone's permission system... right?

Reply:
Certainly not, because I have seen it in many places and many people use it, so not.

Reply:
You don't understand - since essentials.togglejail. is not enabled by default, this will probably prevent people with the old normal permission from using this command without changing their permissions.

Reply:
So I need to add a config option to enable the different jail permissions for togglejail?

Reply:
I'm not seeing much of a way around it... if you come up with a better solution, propose it!

Reply:
should be possible to add this in just using permissions defaults:

- don't check a permission for the command `/togglejail` anymore
- in the `plugin.yml`, have the permission `essentials.togglejail` have the children `essentials.togglejail.*: true` and `essentials.togglejail.offline: false`

that should be backwards compatible

Reply:
thx permission expert zml

Ya if zml says it should work then it should work. That or we can file an issue on PEX!